frontend/src/metabase/visualizations/components/settings/ChartSettingsTableFormatting.unit.spec.js

```diff
@@ -0,0 +1,164 @@
+    );
+    userEvent.click(screen.getByText("Add rule"));
+
+    expect(screen.getByText("String Column")).toBeInTheDocument();
+    expect(screen.getByText(/is equal to toucan/g)).toBeInTheDocument();
+  });
+
+  it("should only let you choose columns of the same type for a rule", () => {
+    setup();
+    expect(screen.getByText("Conditional formatting")).toBeInTheDocument();
+    userEvent.click(screen.getByText("Add a rule"));
+    userEvent.click(screen.getByText("String Column"));
+
+    expect(
+      screen.getByRole("option", { name: "Number Column" }),
+    ).toHaveAttribute("aria-disabled", "true");
+
+    expect(
+      screen.getByRole("option", { name: "Boolean Column" }),
+    ).toHaveAttribute("aria-disabled", "true");
+
+    expect(
+      screen.getByRole("option", { name: "String Column 2" }),
+    ).toHaveAttribute("aria-disabled", "false");
+  });
+
+  describe("should show propriate operators based on column selection", () => {
+    beforeEach(() => {
+      setup();
+      userEvent.click(screen.getByText("Add a rule"));
+    });
+
+    it("string", () => {
+      userEvent.click(screen.getByText("String Column"));
+      //Dismiss Popup
+      userEvent.click(screen.getByText("Which columns should be affected?"));
+
+      userEvent.click(
+        screen.getByTestId("conditional-formatting-value-operator-button"),
+      );
+
+      expect(
+        screen.getByRole("option", { name: "starts with" }),
+      ).toBeInTheDocument();
+
+      expect(
+        screen.getByRole("option", { name: "ends with" }),
+      ).toBeInTheDocument();
+    });
+
+    it("number", () => {
+      userEvent.click(screen.getByText("Number Column"));
+      //Dismiss Popup
+      userEvent.click(screen.getByText("Which columns should be affected?"));
+
+      userEvent.click(
+        screen.getByTestId("conditional-formatting-value-operator-button"),
+      );
+
+      expect(
+        screen.getByRole("option", { name: "is less than" }),
+      ).toBeInTheDocument();
+
+      expect(
+        screen.getByRole("option", { name: "is greater than" }),
+      ).toBeInTheDocument();
+
+      //Quick check for color range option on numberic rules
+      expect(screen.getByText("Formatting style")).toBeInTheDocument();
+      expect(
+        screen.getByRole("radio", { name: /single color/i }),
+      ).toBeChecked();
+      expect(
+        screen.getByRole("radio", { name: /color range/i }),
+      ).not.toBeChecked();
+    });
+
+    it("boolean", () => {
+      userEvent.click(screen.getByText("Boolean Column"));
+      //Dismiss Popup
+      userEvent.click(screen.getByText("Which columns should be affected?"));
+
+      userEvent.click(
+        screen.getByTestId("conditional-formatting-value-operator-button"),
+      );
+
+      expect(
+        screen.getByRole("option", { name: "is true" }),
+      ).toBeInTheDocument();
+
+      expect(
+        screen.getByRole("option", { name: "is false" }),
+      ).toBeInTheDocument();
```

Comment:
Same as above: can we add all of the boolean options to make this spec more complete?

![image](https://github.com/metabase/metabase/assets/22608765/6533dc4e-d1ea-4457-84f0-e36131465749)